src/setupRtl.tsx

```diff
@@ -0,0 +1,45 @@
+import { render, RenderOptions } from "@testing-library/react";
+import React from "react";
+
+import { RemainingPropsAndTestOverrides } from "./types";
+
+/**
+ * Creates a `renderView` function that can be used in unit tests to mount a component.
+ *
+ * @param Component - React component to be rendered.
+ * @param baseProps - Any default props to pass to the component in all tests.
+ * @example
+ * ```tsx
+ * const renderView = setupRtl(ButtonWithText, { onClick: jest.fn() })
+ *
+ * const text = 'Hooray!';
+ * const { props: { onClick }, view } = renderView({ text });
+ *
+ * FireEvent.click(view.getByRole('button'));
+ *
+ * expect(onClick).toHaveBeenCalledWith(text);
+ * ```
+ */
+export function setupRtl<
+  ComponentType extends React.ComponentType,
+  BaseProps extends Partial<React.ComponentProps<ComponentType>>
+>(Component: ComponentType, baseProps?: BaseProps) {
```

Comment:
This came up in some of the emotion feature work.  Is there a way we could specify some global decorators through or in a reusable way here?  This came up in the emotion theme work, where adding context providers every time might be a bit frustrating.  (Redux, ThemeProvider etc).

Not blocking just a thought.

Reply:
I don't see what you mean 😄 but I'll follow up on Slack. Good fodder for a followup issue, I think!

Context for outside folks looking in: this is largely a (now relicensed) copy & paste of what we have internally, rather than net _new_ code.